pkg/sql/conn_executor.go

```diff
@@ -3005,8 +3007,19 @@ func (ex *connExecutor) execCopyIn(
 		}
 	}()
 
+	// We've seen cases where cmd.CopyDone.Done() is called multiple times on
+	// the same wait group which results in a server crash. To prevent this type
+	// of panic from occurring we check whether Done() has already been called
+	// at lease once, and if so, we return an error.
+	if cmd.WaitGroupDone {
```

Comment:
this doesn't seem thread-safe  since checking the boolean needs to be atomic with setting it.

but i think my idea in https://github.com/cockroachdb/cockroach/pull/115712/files#r1417880287 would work to achieve this goal (i.e., use `sync.Once`)